### PR TITLE
add preference for application/json in health request

### PR DESCRIPTION
**Pull request by kevo1ution**
## Changes proposed in this PR

- prefer application/json in health request checks

## Why are we making these changes?

- Deployment errors are given in the ping response

### CODE REVIEW

**Review by shamsimam — COMMENTED** [review]
Can you add an integration test that confirms the `/waiter-ping` is sending back the json response from the backend for a deployment error?

File: waiter/integration/waiter/health_check_test.clj [hunk 1]
```diff
@@ -252,3 +252,25 @@
                 (is (= (str "Hello " (retrieve-username)) (-> backend-response :body str))))))
           (finally
             (delete-token-and-assert waiter-url token)))))))
+
+(deftest ^:parallel ^:integration-fast test-ping-deployment-error
```
Comment (shamsimam) [inline waiter/integration/waiter/health_check_test.clj:256]: Can we introduce assertions in the tests here: https://github.com/twosigma/waiter/blob/master/waiter/integration/waiter/deployment_errors_test.clj

That way we do not have to worry about the overhread of launching a new service and detecting it is failing.

File: waiter/integration/waiter/deployment_errors_test.clj [hunk 2]
```diff
@@ -49,18 +49,19 @@
...
   [response deployment-error]
   `(let [response# ~response
-         body# (:body response#)
-         service-id# (response->service-id response#)]
-     (assert-response-status response# http-503-service-unavailable)
-     (is (str/includes? body# (deployment-error->str ~'waiter-url ~deployment-error))
-         (formatted-service-state ~'waiter-url service-id#))
-     (testing "status is reported as failing"
-       (is
-         (wait-for
```
Comment (shamsimam) [inline waiter/integration/waiter/deployment_errors_test.clj:59]: Are we relying on recent `/waiter-ping` behavior to make sure the `Failing` is returned consistently?

File: waiter/src/waiter/settings.clj [hunk 3]
```diff
@@ -280,7 +281,8 @@
                              :min-hosts 2}
    :entitlement-config {:kind :simple
                         :simple {:factory-fn 'waiter.authorization/->SimpleEntitlementManager}}
-   :health-check-config {:health-check-timeout-ms 200
+   :health-check-config {:health-check-headers-accept "application/json;q=1.0, */*;q=0.9"
```
Comment (shamsimam) [inline waiter/src/waiter/settings.clj:284]: Nitpick: I prefer `health-check-accept-header`

File: waiter/integration/waiter/deployment_errors_test.clj (hunk 2 above)
Comment (kevo1ution) [reply waiter/integration/waiter/deployment_errors_test.clj:59]: @shamsimam yes, should I not be relying on `/waiter-ping`?

Comment (kevo1ution) [reply waiter/integration/waiter/deployment_errors_test.clj:59]: It seems we are relying on `/waiter-ping` endpoint for the json response in these test cases as well

File: waiter/config-full.edn [hunk 4]
```diff
@@ -989,6 +989,9 @@
                            :min-hosts 2}
 
  :health-check-config {
+                       ;; HTTP accept header that is sent to a service when making health check request through /waiter-ping
+                       :health-check-accept-headers "application/json;q=1.0, */*;q=0.9"
```
Comment (shamsimam) [inline waiter/config-full.edn:993]: Nitpick: `health-check-accept-header` (not plural)

File: waiter/integration/waiter/deployment_errors_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/integration/waiter/deployment_errors_test.clj:59]: Previously there used to be a delay in the failing state being reflected. I was checking whether we feel whether the recent /waiter-ping changes mean that we will consistently see the `Failing` state.

Comment (shamsimam) [reply waiter/integration/waiter/deployment_errors_test.clj:59]: This assertion is also testing output from some other endpoint `/apps/<service-id>`
I would check the git history to see why/when this change was introduced.

Comment (kevo1ution) [reply waiter/integration/waiter/deployment_errors_test.clj:59]: Looking at the git history it seems that these were also testing `/apps/<service-id>` endpoint for adding support of the service "status" https://github.com/twosigma/waiter/pull/519 . I'll include this assertion :+1:

Comment (kevo1ution) [reply waiter/integration/waiter/deployment_errors_test.clj]: also /waiter-ping change will wait for "healthy" state with a default timeout of 5 seconds

Comment (shamsimam) [reply waiter/integration/waiter/deployment_errors_test.clj]: okay, that works.

**Review by shamsimam — APPROVED** [review]

**Review by shamsimam — APPROVED** [review]

**Pull request merged**